src/components/PopupWithForm.js

```diff
@@ -45,28 +45,21 @@ export default class PopupWithForm extends Popup {
     const inputValues = this._getInputValues();
     this.renderLoading(true);
     this._handleFormSubmit(inputValues);
-    Promise.resolve(this._handleFormSubmit(inputValues))
-      .then(() => {
-        this._resetForm();
-        this.close();
-      })
-      .catch((err) => {
-        console.error("Error submitting form:", err);
-      })
-      .finally(() => {
-        this.renderLoading(false);
-      });
   }
 
   _setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => this._handleSubmit(e));
   }
 
-  _resetForm() {
+  resetForm() {
     this._popupForm.reset();
   }
 
+  open() {
+    super.open();
+  }
+
   close() {
     super.close();
   }
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.